modules/auto-update/implementation/ot-auto-updater.js

```
@@ -0,0 +1,227 @@
+    readAppVersion() {
+        const file = path.join(appRootPath.path, 'package.json');
+        this.logger.info(`AutoUpdater - Reading app version from ${file}`);
+        const appPackage = fs.readFileSync(file);
+        return JSON.parse(appPackage).version;
+    }
+
+    /**
+     * A promise wrapper for sending a get https requests.
+     * @param {String} url - The Https address to request.
+     * @param {String} options - The request options. 
+     */
+    promiseHttpsRequest(url, options) {
+        return new Promise((resolve, reject) => {
+            const req = https.request(url, options, res => {
+                let body = '';
+                res.on('data', data => {body += data});
+                res.on('end', () => {
+                    if (res.statusCode === 200) return resolve(body);
+                    this.logger.info(`AutoUpdater - Bad Response ${res.statusCode}`);
+                    reject(res.statusCode);
+                });
+            });
+            this.logger.info(`AutoUpdater - Sending request to ${url}`);
+            this.logger.info(`AutoUpdater - Options: ${JSON.stringify(options)}`);
+            req.on('error', reject);
+            req.end();
+        }); 
+    }
+
+    /**
+     * Reads the applications version from the git repository.
+     */
+    async readRemoteVersion() {
+        const options = {}
+        let url =  `${REPOSITORY_URL}/${this.config.branch}/package.json`;
+        if (url.includes('github')) url = url.replace('github.com', 'raw.githubusercontent.com');
+        this.logger.info(`AutoUpdater - Reading remote version from ${url}`);
+        
+        try {
+            const body = await this.promiseHttpsRequest(url, options);
+            const remotePackage = JSON.parse(body);
+            const {version} = remotePackage;
+            return version;
+        } catch(e) {
+            this.logger.error(`This repository requires a token or does not exist. Error message: ${e.message}`);
```

We should probably throw an error here if we are not able to read remote version